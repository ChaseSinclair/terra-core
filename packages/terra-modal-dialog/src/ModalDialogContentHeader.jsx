import React from 'react';
import PropTypes from 'prop-types';
import './ModalDialogContentHeader.scss';

const propTypes = {
  children: PropTypes.node,
};

const defaultProps = {
  children: null,
};

const ModalDialogContentHeader = ({ children, ...customProps }) => (
  <div className={'terra-ModalDialog-contentHeader'} {...customProps} >
    {children}
  </div>
  );

ModalDialogContentHeader.propTypes = propTypes;
ModalDialogContentHeader.defaultProps = defaultProps;

export default ModalDialogContentHeader;
