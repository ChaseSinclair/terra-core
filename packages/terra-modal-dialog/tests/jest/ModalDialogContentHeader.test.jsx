import React from 'react';
import ModalDialogContentHeader from '../../src/ModalDialogContentHeader';

const modalDialogContentHeader = <ModalDialogContentHeader>This is content</ModalDialogContentHeader>;

it('should shallow an open modal', () => {
  const modal = shallow(modalDialogContentHeader);
  expect(modal).toMatchSnapshot();
});

it('should mount an open modal', () => {
  const modal = mount(modalDialogContentHeader);
  expect(modal).toMatchSnapshot();
});

it('should render an open modal', () => {
  const modal = render(modalDialogContentHeader);
  expect(modal).toMatchSnapshot();
});

it('should have class terra-ModalDialog-contentHeader', () => {
  const modal = shallow(modalDialogContentHeader);
  expect(modal.prop('className')).toEqual('terra-ModalDialog-contentHeader');
});
