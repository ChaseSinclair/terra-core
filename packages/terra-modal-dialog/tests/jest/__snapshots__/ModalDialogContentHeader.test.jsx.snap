exports[`test should mount an open modal 1`] = `
<ModalDialogContentHeader>
  <div
    className="terra-ModalDialog-contentHeader">
    This is content
  </div>
</ModalDialogContentHeader>
`;

exports[`test should render an open modal 1`] = `
<div
  class="terra-ModalDialog-contentHeader">
  This is content
</div>
`;

exports[`test should shallow an open modal 1`] = `
<div
  className="terra-ModalDialog-contentHeader">
  This is content
</div>
`;
